src/iexamine.cpp

```diff
@@ -3766,20 +3787,28 @@ static void pick_plant( Character &you, const tripoint &examp,
         return;
     }
 
-    const float survival = you.get_skill_level( skill_survival );
-    you.practice( skill_survival, 6 );
+    const int survival = you.get_skill_level( skill_survival );
+    ///\EFFECT_INT Intelligence caps survival skill gains from foraging
+    const int max_forage_skill = you.int_cur / 5 + 1;
+    ///\EFFECT_SURVIVAL decreases survival skill gain from foraging (NEGATIVE)
+    const int max_exp = ( max_forage_skill - survival );
+    // Award experience for foraging attempt regardless of success
+    you.practice( skill_survival, rng( 1, max_exp ), max_forage_skill );
```

Comment:
This ended up duplicated with minor changes in enough places that I will probably move it out to a helper before un-drafting this PR.